storage/src/digest.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use std::{fmt, io, ops::Deref};
+
+use smallvec::SmallVec;
+use snarkvm_utilities::{ToBytes, Write};
+
+type InnerType = SmallVec<[u8; 64]>;
```

Comment:
Can the length of a digest be greater than `64`?

Reply:
in testnet1? not as far as i've observed. `smallvec` crate will allocate to heap if > 64 bytes in length. the intent of the digest type is to make this easier to tweak for performance.

Reply:
Now that I've seen the rest of the changes I guess the use of `SmallVec` is just for ergonomics, i.e. to know the length of the digest etc. rather than consider any memory implications.

Reply:
no. `SmallVec` avoids heap allocations in reality, while allowing digests to be larger to accommodate other network schemes without substantial modification (basically to decouple storage from network scheme).

Reply:
Yeah, I meant right now, while all the digests we're using are <=64B.